pkg/kritis/crd/securitypolicy/securitypolicy.go

```diff
@@ -48,17 +49,24 @@ func ImageSecurityPolicies() ([]v1beta1.ImageSecurityPolicy, error) {
 
 // ValidateImageSecurityPolicy checks if an image satisfies ISP requirements
 // It returns a list of vulnerabilites that don't pass
-func ValidateImageSecurityPolicy(isp v1beta1.ImageSecurityPolicy, project, image string, client metadata.MetadataFetcher) ([]metadata.Vulnerability, error) {
+func ValidateImageSecurityPolicy(isp v1beta1.ImageSecurityPolicy, project, image string, client metadata.MetadataFetcher) ([]SecurityPolicyViolation, error) {
```

Comment:
hmm, as a further refactor would it make sense to pass in a name.Digest https://github.com/google/go-containerregistry/blob/master/pkg/name/digest.go#L30 instead of a string? That way we don't have to keep checking that things are fully resolved everywhere.

Reply:
So right now we do the check in two places:

1. For the cron job, I like keeping the check here in ValidateImageSecurityPolicy because we also get information about CVE violations, and all of that gets added as an annotation to the pod. If we did the check before calling this, we wouldn't get that.

2. In the webhook we do it before we even check against security policies because it results in an automatic denial